prov/sockets/src/sock_cntr.c

```diff
@@ -292,20 +295,36 @@ static int sock_cntr_wait(struct fid_cntr *cntr, uint64_t threshold,
 		while (atomic_get(&_cntr->value) < threshold) {
 			sock_cntr_progress(_cntr);
 			if (timeout >= 0 && fi_gettime_ms() >= end_ms) {
-				ret = FI_ETIMEDOUT;
+				ret = -FI_ETIMEDOUT;
 				break;
 			}
 		}
 		pthread_mutex_lock(&_cntr->mut);
 	} else {
-		ret = fi_wait_cond(&_cntr->cond, &_cntr->mut, timeout);
+
+		remaining_ms = timeout;
+		if (timeout >= 0) {
+			start_ms = fi_gettime_ms();
+			end_ms = start_ms + timeout;
```

Comment:
We are doing in the if - true case. We can just move this up before the if statement.